libs/utils/android/system.py

```diff
@@ -464,4 +465,31 @@ def packages_info(target, apk_filter=''):
             return packages
         return None
 
+    @staticmethod
+    def grant_permissions(target, package):
+        """
+        Grant a package all the permissions it may ask for
+
+        :param target: instance of devlib Android target
+        :type target: devlib.target.AndroidTarget
+
+        :param package: package name (i.e. com.foo.bar)
+        :type package: str
+        """
+        
+        dumpsys = target.execute('dumpsys package {}'.format(package))
+
+        permissions = re.search(
+            'requested permissions:\s*(?P<permissions>(android.permission.+\s*)+)', dumpsys
+        )
+
+        permissions = permissions.group('permissions').replace(" ", "").splitlines()
+        
+        for permission in permissions:
+            try:
+                target.execute('pm grant {} {}'.format(package, permission))
+            except:
+                # Some permissions cannot be granted, ignore them
```

Comment:
Does the app run if you don't grant a permission?

Reply:
Yes, you just get stuck with a permission request overlayed on top of the app.

Reply:
OK, then if you add this to `devlib` it's probably good to log a debug message saying that a certain permission wasn't granted.